aws-sdk-go/LeoFSTest.go

```diff
@@ -0,0 +1,485 @@
+            break
+        } else {
+            marker = *res.NextMarker
+        }
+    }
+    log.Println("===== End =====")
+    if count != total {
+        log.Fatalln("Number of Objects NOT Match!")
+    }
+    log.Println("===== Multiple Page Lsit Objects End =====")
+    log.Println()
+}
+
+func multiDelete(bucketName, prefix string, total int) {
+    log.Printf("===== Multiple Delete Objects [%s/%s] Start =====\n", bucketName, prefix)
+    bucketStr := aws.String(bucketName)
+    delKeyList := make([]*s3.ObjectIdentifier, total)
+    for i := 0; i < total; i++ {
+        delKeyList[i] = &s3.ObjectIdentifier{
+            Key : aws.String(fmt.Sprintf("%s%d", prefix, i)),
+        }
+    }
+    res, err := svc.DeleteObjects(&s3.DeleteObjectsInput{
+        Bucket : bucketStr,
+        Delete : &s3.Delete{
+            Objects : delKeyList,
+        },
+    })
+    if err != nil {
+        log.Fatalln(err)
+    }
+    for _, obj := range res.Deleted {
+        log.Printf("Deleted %s/%s", bucketName, *obj.Key)
+    }
+    if len(res.Deleted) != total {
+        log.Fatalln("Number of Objects NOT Match!")
+    }
+    log.Println("===== Multiple Delete Objects End =====")
+    log.Println()
+}
+
+func setBucketAcl(bucketName, permission string) {
+    log.Printf("===== Set Bucket ACL [%s] (%s) Start =====\n", bucketName , permission)
+    var checkList []string
+    if permission == "private" {
+        checkList = []string{"FULL_CONTROL"}
+    } else if permission == "public-read" {
+        checkList = []string{"READ", "READ_ACP"}
+    } else if permission == "public-read-write" {
+        checkList = []string{"READ", "READ_ACP", "WRITE", "WRITE_ACP"}
+    } else {
+        log.Fatalln("Invalid Permission!")
+    }
+    bucketStr := aws.String(bucketName)
+    _, err := svc.PutBucketACL(&s3.PutBucketACLInput{
+        Bucket  : bucketStr,
+        ACL     : aws.String(permission),
+    })
+    if err != nil {
+        log.Fatalln(err)
+    }
+    res, err := svc.GetBucketACL(&s3.GetBucketACLInput{
+        Bucket  : bucketStr,
+    })
+    if err != nil {
+        log.Fatalln(err)
+    }
+    log.Printf("Owner ID: S3Owner [name=%s,id=%s]\n", *res.Owner.DisplayName, *res.Owner.ID)
+    list := make([]string, len(res.Grants))
+    for i, grant := range res.Grants {
+        log.Printf("Grantee: %s \t Permissions: %s", *grant.Grantee.URI, *grant.Permission)
+        list[i] = *grant.Permission
+    }
+    for _, item := range checkList {
+        found := false
+        for _, ele := range list {
+            if ele == item {
+                found = true
+                break
+            }
+        }
+        if !found {
+            log.Fatalln("ACL NOT Match!")
+        }
+    }
+}
+
+//Ref: http://play.golang.org/p/S2s3e0nj4s
+func doesFileMatch(r1, r2 io.Reader) bool {
+    buf1 := bufio.NewReader(r1)
+    buf2 := bufio.NewReader(r2)
+    for {
+        b1, err1 := buf1.ReadByte()
+        b2, err2 := buf2.ReadByte()
```

Comment:
Using `Read(p []byte) (n int, err error)` and compare multiple bytes at once by using http://golang.org/pkg/bytes/#Compare seems to be more efficient.